File: tenant-profile-service/src/main/java/com/stackroute/tenantprofileservice/model/TenantProfile.java
package com.stackroute.tenantprofileservice.model;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.stereotype.Component;

import javax.validation.constraints.Email;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.util.List;

/**
 * @Author Siva
 * @Date 11/18/2021 2:18 PM
 */
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@ToString
@Component
@Document
public class TenantProfile  {
    @Id
    @Email
    private String email;

//    @NotNull
//    @Size(min = 4, max = 20)
    private String fullName;

//    @NotNull
//    @Min(value = 18)
    private int age;

  //  @NotNull
    private String city;

 //   @NotNull
    private String contactNumber;

  //  @NotNull
    private Gender gender;

  //  @NotNull
    private String state;

  //  @NotNull
    private List<String> hobbies;

  //  @NotNull
    private String ambition;

   // @NotNull
    private String passion;

  //  @NotNull
    private Questionaire questionaire;

  //  @NotNull
    private School school;

   // @NotNull
    private College college;

   // @NotNull
    private Career career;

    private byte[] image;

}

File: product-webapp/webapp/src/app/model/tenant-profile.ts
import { Career } from "./career";
import { College } from "./college";
import { Gender } from "./gender.enum";
import { Questionaire } from "./questionaire";
import { School } from "./school";

export class TenantProfile {

    email: String;
    fullName: String;
    age: Number;
    city: String;
    contactNumber: String;
    gender: Gender;
    state: String;
    hobbies: String[];
    ambition: String;
    passion: String;
    questionaire: Questionaire;
    school: School;
    college: College;
    career: Career;
    image: any;
}
